spikeinterface/preprocessing/remove_bad_channels.py

```diff
@@ -1,50 +1,280 @@
+
+        # Take the mode of the chunk estimates as final result. Convert to binary good / bad channel output.
+        bad_channel_labels, _ = scipy.stats.mode(channel_labels, axis=1, keepdims=False)
+        if order_r is not None:
+            bad_channel_labels = bad_channel_labels[order_r]
+
+        bad_inds, = np.where(bad_channel_labels != 0)
+        bad_channel_ids = recording.channel_ids[bad_inds]
+
+        if bad_channel_ids.size > recording.get_num_channels() / 3:
+            warnings.warn("Over 1/3 of channels are detected as bad. In the precense of a high"
+                          "number of dead / noisy channels, bad channel detection may fail "
+                          "(erroneously label good channels as dead).")
+
+    return bad_channel_ids, bad_channel_labels
+
+
+# ----------------------------------------------------------------------------------------------
+# IBL Detect Bad Channels
+# ----------------------------------------------------------------------------------------------
 
-        parents_chan_ids = recording.get_channel_ids()
-        channel_ids = parents_chan_ids[keep_inds]
-        self._parent_channel_indices = recording.ids_to_indices(channel_ids)
+def detect_bad_channels_ibl(raw, fs, psd_hf_threshold,
+                            dead_channel_thr=-0.5,
+                            noisy_channel_thr=1.,
+                            outside_channel_thr=-0.75,
+                            n_neighbors=11,
+                            nyquist_threshold=0.8):
+    """
+    Bad channels detection for Neuropixel probes developed by IBL
+
+    Parameters
+    ----------
+    raw : traces
+        (num_samples, n_channels) raw traces
+    fs : float
+        sampling frequency
+    psd_hf_threshold : float
+        Threshold for high frequency PSD. If mean PSD above `nyquist_threshold` * fn is greater than this 
+        value, channels are flagged as noisy (together with channel coherence condition).
+    dead_channel_thr : float, optional
+        Threshold for channel coherence below which channels are labeled as dead, by default -0.5
+    noisy_channel_thr : float
+        Threshold for channel coherence above which channels are labeled as noisy (together with psd condition),
+        by default -0.5
+    outside_channel_thr : float
+        Threshold for channel coherence above which channels
+    n_neighbors : int, optional
+        Number of neighbors to compute median fitler, by default 11
+    nyquist_threshold : float, optional
+        Threshold on Nyquist frequency to calculate HF noise band, by default 0.8
+
+    Returns
+    -------
+    1d array
+        Channels labels: 0: good,  1: dead low coherence / amplitude, 2: noisy, 3: outside of the brain
+    """
+    _, nc = raw.shape
+    raw = raw - np.mean(raw, axis=0)[np.newaxis, :]
+    fscale, psd = scipy.signal.welch(raw, fs=fs, axis=0)
+
+    xcor, xcorr_neighbors, xcorr_distant = compute_correlations(raw, n_neighbors)
```

Comment:
`compute_correlations` is called only once.
`channels_similarity`is also called once.
This make the maths hard to follow.
I would propose to make a linear code without sub function calls because they are call once.
no ?

Reply:
done